README.md

```diff
@@ -10,399 +10,451 @@ Welcome to A December of Algorithms. This is a small collection of algorithms to
-  - `Note #1:` OpenWeatherMap APIs need a valid API key to allow responses, Packages won't work if you don't provide one. You can signup for a free API key on the OpenWeatherMap website.
-  - `Note #2:` API keys are private and **should not be made public**. When committing your code, please make sure you replace your original API key with a dummy key.
-  - `Note #2:`Alternatively, you can look into saving your API key as an Environment Variable.
-  One final alternative is to simply store the API key in a text file and add that file to your `.gitignore` file. When you need to access it, open the file using your program and retrieve the API key.
-
-
-- **Sample Output**
-  ```
-  Taken at 20:35 (IST), 04-12-2018
-  Current Temperature difference between Chennai,TN and London,GB is 19°C
-  ```
+### **December 17 - Magic Squares**
+  - **Problem**
+    - A magic square of order n is an arrangement of ``n^2`` numbers, containing distinct integers from ``1`` to ``n^2``, in a square, such that the ``n`` numbers in all rows, all columns, and both diagonals sum to the same constant.
+    - This constant is called the magic constant or magic sum, M, which is given by:
+        ```math
+        M = n(n^2+1)/2
+        ```
+    - Now the problem is to generate all the possible magic squares if the order of the square is given as input.
+
+
+  - **Example**
+   <img src="/src/docs/Dec17-Example.png" width="375" height="288"/>
 
-- **Resources:**
-  - [How to get your API key from OpenWeatherMap](https://openweathermap.org/appid)
-  - [How to retrieve weather data for a city](https://openweathermap.org/current#name)
+  - **Resources**
+    - [Magic Squares #1](http://www.math.wichita.edu/~richardson/mathematics/magic%20squaresorder3magicsquare.html)
+    - [Magic Squares #2](https://mindyourdecisions.com/blog/2015/11/08/how-many-3x3-magic-squares-are-there-sunday-puzzle/)
 
 
+### **December 18 - How Secure is your Password?**
+  - **Problem**
+    - Brute Force Attack is the most widely known password cracking method. This attack simply tries to use every possible printable ASCII character combination as a password. Though brute forcing can be a little exhaustive, the likelihood of a password being cracked is made certain.
+    - The problem is to find the approximate time that will be take to crack a given password using Brute Force Attack. The larger the time, the stronger the password is.
+    - Assume **500,000** passwords can be tested per second.
+    - `Optional:` Perform a check for dictionary attack before checking for brute-force by directly comparing the input password with each password in the **.txt** file in the reference.
+
+
+  - **Examples**
+    ```
+    Enter Password: abc
+    Time Taken to brute (approx.): Instantly
+    ```
+    ```
+    Enter Password: h3lL0
+    Time Taken to brute (approx.): 18000 sceonds
```

Comment:
Are you sure your calculations are right?

Assuming numbers, upper & lower case characters and 10 special characters, that's 10 + 26 +26 + 10 = 72 possible characters and 500,000 passwords/sec.

| Length of Character | Number of Combinations | Time to Bruteforce |
| --- | --- | --- |
| 1 | 72 | <1s |
| 2 | 5,184 | <1s |
| 3 | 373,284 | <1s |
| 4 | 26,873,856 | 53.74s |
| 5 | 1,934,898,192 | 3869.79s |
| 6 | 139,314,069,504 | 278628.13s |

Reply:
My bad. I've taken the Full extended ASCII Table into consideration rather than just the printables :3
I'll update the outputs with calculations for all **ASCII printable characters (character code 32-126)**